### PR TITLE
HexColor: Added rule

### PR DESCRIPTION
**Pull request by DaveStein**

### CODE REVIEW

File: src/Validation/Rules/HexColorRule.php [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Behance\NBD\Validation\Rules;
+
+use Behance\NBD\Validation\Abstracts\RegexRuleAbstract;
+
+/**
+ * Represents the container of a rule to be "run"
```
Comment (bryanlatten) [inline src/Validation/Rules/HexColorRule.php]: quickly explain what this is

**Comment by kgeorgiev** [comment]
:+1:

File: src/Validation/README.md [hunk 2]
```diff
@@ -97,6 +97,7 @@ else {
 * json
 * notEmpty
 * url
+* hexColor
```
Comment (jimdelois) [inline src/Validation/README.md]: Potentially worth explaining inline that the accepts either 3 or 6 digits, without the leading `#` ?

**Comment by bryanlatten** [comment]
:+1:

File: tests/Validation/Rules/HexColorRuleTest.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        [ 'A000AG', false ],
+        [ 'ábč', false ],
+        [ 'ábčabc', false ],
+        [ 'ÁBČabc', false ],
+        [ 123.0e26, false ],
+        [ null, false ],
+        [ false, false ],
+        [ true, false ],
+        [ ( new \stdClass() ), false ],
+        [ ( function() {} ), false ],
```
Comment (jimdelois) [inline tests/Validation/Rules/HexColorRuleTest.php]: Align these bad boys.  Also consider adding test cases for leading/trailing spaces showing that they fail?  No biggie.

File: src/Validation/README.md (hunk 2 above)
Comment (DaveStein) [inline src/Validation/README.md]: 3 or 6 is implied. Without the leading is in the error message.

**Comment by jimdelois** [comment]
:+1:

File: tests/Validation/Rules/HexColorRuleTest.php (hunk 3 above)
Comment (bryanlatten) [inline tests/Validation/Rules/HexColorRuleTest.php]: oh yeah, and the #!

Comment (DaveStein) [inline tests/Validation/Rules/HexColorRuleTest.php]: added tests.

**Pull request merged**